pkg/util/crypt/crypt_dev.go

```diff
@@ -222,13 +210,14 @@ func copyDeviceContents(source, dest string, size int64) error {
 	for writtenSoFar < size {
 		numRead, err := syscall.Read(sourceFd, buffer)
 		if err != nil {
-			return fmt.Errorf("Unable to read the the file %s", source)
+			return fmt.Errorf("unable to read the the file: %s: %s", source, err)
 		}
 		numWritten, err := syscall.Write(destFd, buffer)
 		if err != nil {
-			return fmt.Errorf("Unable to write to destination %s", dest)
+			return fmt.Errorf("unable to write to destination: %s: %s", dest, err)
 		}
 
+		// TODO: is this really necessary???
 		if numRead != numWritten {
 			sylog.Debugf("numRead != numWritten")
 		}
```

Comment:
@schebro, is this necessary? Can it be removed now?

Reply:
I added it so that we see when there is some discrepancy b/w read and written data. Not strictly necessary. But, I prefer to leave it in for debugging purposes.